content/appendices.md

```diff
@@ -611,10 +611,18 @@ characters permitted in user ID localparts. There are currently active
 users whose user IDs do not conform to the permitted character set, and
 a number of rooms whose history includes events with a `sender` which
 does not conform. In order to handle these rooms successfully, clients
-and servers MUST accept user IDs with localparts from the expanded
-character set:
+and servers MUST accept user IDs with localparts consisting of any legal
+unicode codepoint except for `:` and `NUL` (U+0000), including other control
+characters and the empty string. Localparts MUST be valid UTF-8 sequences.
```

Comment:
we're talking characters rather than bytes here, so "MUST be valid UTF-8 sequences" seems confusing. I think we want:

```suggestion
and servers MUST accept user IDs with localparts consisting of any legal
non-surrogate Unicode code points except for `:` and `NUL` (U+0000), including other control
characters and the empty string.
```

Reply:
I did a bit of testing on Synapse. As far as I can tell, it rejects any attempt to use the surrogates. If you encode the surrogates as UTF-8 (`<ed> <a0> <80>`, say, which is the UTF-8 encoding of U+D800), and send that over HTTP, Synapse will reject the request when it tries to decode the UTF-8. 

Likewise, it refuses to persist any strings containing the surrogates to the database, or to put them in outgoing HTTP requests/responses, because it fails to encode them as UTF-8.

I'm therefore pretty happy to say that nobody has to deal with the surrogates.